source/server/hot_restarting_child.cc

```diff
@@ -19,13 +21,15 @@ int HotRestartingChild::duplicateParentListenSocket(const std::string& address)
     return -1;
   }
 
-  RpcGetListenSocketRequest rpc;
-  ASSERT(address.length() < sizeof(rpc.address_));
-  StringUtil::strlcpy(rpc.address_, address.c_str(), sizeof(rpc.address_));
-  sendMessage(parent_address_, rpc);
-  RpcGetListenSocketReply* reply =
-      receiveTypedRpc<RpcGetListenSocketReply, RpcMessageType::GetListenSocketReply>();
-  return reply->fd_;
+  HotRestartMessage wrapped_request;
+  wrapped_request.mutable_pass_listen_socket_request()->set_address(address);
+  sendHotRestartMessage(parent_address_, wrapped_request);
+
+  std::unique_ptr<HotRestartMessage> wrapped_reply = receiveHotRestartMessage(/*blocking=*/true);
```

Comment:
nit: for these type of bool params, you might consider a 2 value enum which is easier to read at call sites.

Reply:
Done.